analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnModulus.cs

```diff
@@ -34,6 +35,14 @@ public sealed class EqualityOnModulus : SonarDiagnosticAnalyzer
         private const string DiagnosticId = "S2197";
         private const string MessageFormat = "The result of this modulus operation may not be {0}.";
 
+        private static readonly string[] MethodsOrPropertiesReturningAlwaysPositiveInt =
+        {
+            "System.Array.Length",
+            "System.Collections.Generic.IEnumerable<int>.Count<int>()",
```

Comment:
It should not raise for any IEnumerable, not just the one of integers.

Reply:
Rules S1155 and S3981 check similar things and might contain some things that could help you..

Reply:
Note: Rule S1155 is currently in the process of being refactored (to support VB.NET as well) #5106